contracts/AnchoredView/AnchoredView.sol

```diff
@@ -143,45 +143,33 @@ contract AnchoredView is SymbolConfiguration {
      * @notice Returns price denominated in USD, with 6 decimals
      * @dev If price was posted by source, return it. Otherwise, return anchor price converted through source ETH price.
      */
-    function prices(string calldata symbol) external view returns (uint256) {
-        uint256 price = _prices[symbol];
+    function prices(string calldata symbol) external view returns (uint) {
+        uint price = _prices[symbol];
 
         if (price != 0) {
             return price;
         } else {
-            uint256 usdPerEth = _prices["ETH"];
-            uint256 ethPerToken = readAnchor(getCTokenAddress(symbol));
+            uint usdPerEth = _prices["ETH"];
+            require(usdPerEth > 0, "eth price not set, cannot convert eth to dollars");
+
+            CTokenMetadata memory tokenConfig = getCTokenConfig(symbol);
+            uint ethPerToken = readAnchor(tokenConfig);
 
-            // ethPerToken has 18 decimals since the usdt, usdc, wbtc tokens hit
-            // usdPerEth has 6 decimals
-            // scaling by 1e18 and dividing leaves 1e6, as desired
-            return mul(usdPerEth, 1e18) / ethPerToken;
+            return mul(usdPerEth, ethPerToken) / tokenConfig.anchorAdditionalScale;
         }
     }
 
     /**
      * @dev fetch price in eth from proxy and convert to usd price using anchor usdc price.
      * @dev Anchor usdc price has 30 decimals, and anchor general price has 18 decimals, so multiplying 1e18 by 1e18 and dividing by 1e30 yields 1e6
      */
-    function getAnchorInUsd(address tokenAddress, uint256 usdcPrice) public view returns (uint256) {
-        // TODO: can get rid of trhis if handle decimals more elegantly
-        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
-            // hard code to 1 dollar
-            return oneDollar;
-        }
-
-        uint priceInEth = readAnchor(tokenAddress);
-        uint additionalScale;
-        if ( tokenAddress == cWbtcAddress ){
-            // wbtc proxy price is scaled 1e(36 - 8) = 1e28, so we need 8 more to get to 36
-            additionalScale = 1e8;
-        } else {
-            // all other tokens are scaled 1e18, so we need 18 more to get to 36
-            additionalScale = 1e18;
-        }
+    function getAnchorInUsd(address cToken, uint ethPerUsdc) public view returns (uint) {
+        if (cToken == cUsdcAddress || cToken == cUsdtAddress) return 1e6;
```

Comment:
Isn't this just because the tokens have 6 decimals, I think the comment might make it more confusing?

Reply:
this is b/c cusdc and cusdt never change, In reality this won't ever get called internally with cusdc and cusdt. We could actually delete this line, actually it should always return 1e6 anwyays after running the math for real. 

Also, it could be internal ...

Reply:
or if you mean the comment below, we are converting through the current price oracle eth usd price, and it has 30 decimals. Whatever the numerator is, we need to make sure it has 36 decimals before dividing, ending with the 6 we want

Reply:
maybe oneDollar constant instead of 1e6 for clarity?